### PR TITLE
Reporters

### PR DESCRIPTION
**Pull request by shweta97**
Created a separate reporters folder and made two kinds of reporters: color and plain. Changed the class Reporters to use colorama to make ANSI sequences work on Windows. Fixes issue #33 and #34.

### CODE REVIEW

File: pyta.py [hunk 1]
```diff
@@ -10,24 +10,25 @@
...
 import os
 from urllib.request import pathname2url
 
+
 # Local version of website; will be updated later.
 HELP_URL = 'file:' + pathname2url(os.path.abspath('website/index.html'))
 
 
-def check(module_name):
+def check(module_name, reporter=ColorReporter()):
```
Comment (david-yz-liu) [inline pyta.py]: Don't pass in an instance of the class, pass in the class itself `reporter=ColorReporter` and then call it to create a new reporter in the body of this function.

This is precisely the "mutable default argument" problem.

File: reporters/color_reporter.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    def print_message_ids(self):
+        # Sort the messages by their type.
+        self._messages.sort(key=lambda s: s[0])
+        # Check if the OS currently running is Windows
+
+        init(wrap=(sys.platform == "win32"))
+        for msg in self._messages:
+            if msg.msg_id.startswith('E'):
+                # Error codes appear in red
+                # code = '\033[1;31m' + msg.msg_id + '\033[0m:'
```
Comment (david-yz-liu) [inline reporters/color_reporter.py]: You can delete this now (and the comment below). Our goal is to hide what the ANSI codes actually are.

File: reporters/plain_reporter.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        self._messages.append(msg)
+
+    def print_message_ids(self):
+        # Sort the messages by their type.
+        self._messages.sort(key=lambda s: s[0])
+        # Check if the OS currently running is Windows
+
+        for msg in self._messages:
+            if msg.msg_id.startswith('E'):
+                # Error codes appear in red
```
Comment (david-yz-liu) [inline reporters/plain_reporter.py]: You can delete these comments as well.

**Comment by david-yz-liu** [comment]
@shweta97 I see you've fixed both issues in this pull request. When you add a fix to another issue in the same pull request (which should be rare), please update the description to reflect this new issue fix.

File: pyta.py (hunk 1 above)
Comment (shweta97) [inline pyta.py]: Can i call it like this 'reporter()' to create a new instance?

Comment (david-yz-liu) [inline pyta.py]: @shweta97 we resolved this in the meeting last week. You should update your pull request accordingly (and do another pull from master to resolve conflicts).

**Comment by shweta97** [comment]
@david-yz-liu: I have pulled from master and updated the function check() according to our discussion last meeting. I have also made the colour reporter classes inherit from the plain reporter class.

File: pyta.py [hunk 4]
```diff
@@ -10,13 +10,14 @@
...
 import pylint.lint as lint
-from pylint.reporters import BaseReporter
+from reporters.color_reporter import ColorReporter
 from astroid import MANAGER
 
 import webbrowser
 import sys
 
 
+
```
Comment (david-yz-liu) [inline pyta.py]: Delete this line. Don't submit whitespace changes in your pull requests. You should be checking for this before committing your work.

File: reporters/color_reporter.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+
+class ColorReporter(PlainReporter):
+    def __init__(self):
+        super().__init__()
+
+    # Override this method
+    def print_message_ids(self):
+        # Check if the OS currently running is Windows
+        init(wrap=(sys.platform == "win32"))
```
Comment (david-yz-liu) [inline reporters/color_reporter.py]: Please use single-quotes for string literals

**Pull request merged**